lib/cloudregister/registerutils.py

```diff
@@ -52,18 +52,17 @@ def add_hosts_entry(smt_server):
 
     smt_hosts_entry_comment = '\n# Added by SMT registration do not remove, '
     smt_hosts_entry_comment += 'retain comment as well\n'
-    hosts = open('/etc/hosts', 'a')
-    hosts.write(smt_hosts_entry_comment)
-    smt_ip = smt_server.get_ipv4()
-    if has_ipv6_access(smt_server):
-        smt_ip = smt_server.get_ipv6()
-    entry = '%s\t%s\t%s\n' % (
-        smt_ip,
-        smt_server.get_FQDN(),
-        smt_server.get_name()
-    )
-    hosts.write(entry)
-    hosts.close()
+    with open('/etc/hosts', 'a') as hosts_file:
+        hosts_file.write(smt_hosts_entry_comment)
+        smt_ip = smt_server.get_ipv4()
+        if has_ipv6_access(smt_server):
+            smt_ip = smt_server.get_ipv6()
+        entry = '%s\t%s\t%s\n' % (
+            smt_ip,
+            smt_server.get_FQDN(),
+            smt_server.get_name()
+        )
+        hosts_file.write(entry)
```

Comment:
If you are using context manager, I know calling close() is an extra line of code ;), then the creation of the string we write should be outside, i.e.

```
smt_ip = smt_server.get_ipv4()
if has_ipv6_access(smt_server):
    smt_ip = smt_server.get_ipv6()
entry = '%s\t%s\t%s\n' % (
    smt_ip,
    smt_server.get_FQDN(),
    smt_server.get_name()
)
with open('/etc/hosts', 'a') as hosts_file:
    hosts_file.write(smt_hosts_entry_comment)
    hosts_file.write(entry)
```